src/Microsoft.Health.Fhir.SqlServer/Features/ChangeFeed/SqlServerFhirResourceChangeDataStore.cs

```diff
@@ -98,49 +98,43 @@ public async Task<IReadOnlyCollection<ResourceChangeData>> GetRecordsAsync(long
                 // The GetRecordsAsync function would be called every second by one agent.
                 // So, it would be a good option that opens and closes a connection for each call,
                 // and there is no database connection pooling in the Application at this time.
-                using (SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync(cancellationToken: cancellationToken))
+                using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
+                using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
                 {
-                    await sqlConnection.OpenAsync(cancellationToken);
                     if (ResourceTypeIdToTypeNameMap.IsEmpty)
                     {
                         lock (ResourceTypeIdToTypeNameMap)
                         {
-                            if (ResourceTypeIdToTypeNameMap.IsEmpty)
```

Comment:
Due to multithread, we need to check the state of ResourceTypeIdToTypeNameMap again after entering the lock statement's block.

Reply:
Good catch! I think I took it out by mistake. Readded the check